app/controllers/pets_controller.rb

```diff
@@ -67,4 +77,18 @@ def pet_params
                   :sex)
   end
 
+  def delete_pet_apps(pet)
+    pet.apps.each { | app | pet.apps.delete(app) } if pet.apps.present?
+  end
```

Comment:
I think this logic should probably be executed on the model level. So in the Pet model, have the method 
```ruby 
def delete_apps 
  apps.destroy_all if apps.present?
end 
```
Then in the `destroy` method, we can just call `pet.delete_apps`

Reply:
But also, not entirely sure we need any of this logic.... won't just including a `dependent: :destroy` on the `has_many :pet_apps` relationship take care of destroying associated apps?

Reply:
Ah yes. I keep forgetting about that 🙃